text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -1,77 +1,99 @@
+### Random Value (R)
 
-The `R` value MUST be derived as follows:
+A common random value (that is known or propagated to all participants) is the main ingredient that enables consistent probability sampling. Each participant can compare this value (R) with their rejection threshold (T) to make a consistent sampling decision across an entire trace (or even across a group of traces).
 
-* If the key `rv` is present in the Tracestate header, then `R = rv`.
-* Else if the Random Trace ID Flag is `true` in the traceparent header, then `R` is the lowest-order 56 bits of the trace-id.
-* Else `R` MUST be generated as a random value in the range `[0, (2**56)-1]` and added to the Tracestate header with key `rv`.
+This proposal supports two sources of randomness:
+
+- **A custom source of randomness**: This proposal allows for a _random_ (or pseudo-random) 56-bit value. We refer to this as `rv` (this was previously known as `r-value`). This can be generated and propagated through the `tracestate` header and the tracestate attribute in each span.
+- **Using TraceID as a source of randomness**: This proposal introduces using the last 56 bits of the `traceid` as the source of randomness. This can be done if the root participant knows that the `traceid` has been generated in a random or pseudo-random manner.
+
+### Consistent Sampling Decision Approach
+
+Given the above building blocks, let's look at how a participant can make consistent sampling decisions. For this, two values MUST be present in the `SpanContext`:
+
+1. The common source of randomness: the 56-bit `R` value.
+2. The rejection threshold: the 56-bit `T` value.
+
+If `R` >= `T`, *keep* the span, else *drop* the span.
+
+`T` represents the maximum threshold that was applied in all previous consistent sampling stages. If the current sampling stage applies a greater threshold value than any stage before, it MUST update (increase) the threshold correspondingly.
+
+## Explanation
+
+### Example
+
+Here is an example involving three participating operations `A`, `B`, and `C`:
+
+`A` -> `B` -> `C`
+
+where -> indicates a parent to child relationship.
+
+- `A` uses consistent probability sampling with a sampling probability of 0.25. This corresponds to a rejection threshold of (1-0.25)*2^56 = 0xc0000000000000.
+- `B` uses consistent probability sampling with a sampling probability of 0.5. This corresponds to a rejection threshold of (1-0.5)*2^56 = 0x80000000000000.
+- `C` uses a parent-based sampler. This means that C indirectly follows the same rejection threshold as its parent B.
+
+When the sampling decision for `A` is to *keep* the span, its outgoing traceparent will have the `sampled` flag SET and the `th` in its outgoing tracestate will be set to `0xc0_0000_0000_0000`.
+When the sampling decision for `A` is to *drop* the span, its outgoing traceparent will have the `sampled` flag UNSET but the `th` in its outgoing tracestate will still be set to `0xc0_0000_0000_0000`.
```

Comment:
```suggestion
When the sampling decision for `A` is to *drop* the span, its outgoing traceparent will have the `sampled` flag UNSET and `th` in its outgoing tracestate will be removed.
```

Shouldn't the `th` value be removed in this case? There is no need to propagate the `th`-value if the sampling decision is negative. Otherwise, we would have to define a way to encode a threshold of 2^56 corresponding to a 0% probability.